test/datatable.jl

```diff
@@ -112,48 +109,49 @@ module TestDataTable
     @test isequal(dt, DataTable(a=[1, 2], b=["a", "b"], c=[:c, :d]))
 
     #test_group("Empty DataTable constructors")
-    dt = DataTable(Nullable{Int}, 10, 3)
+    dt = DataTable(?Int, 10, 3)
     @test size(dt, 1) == 10
     @test size(dt, 2) == 3
-    @test typeof(dt[:, 1]) == NullableVector{Int}
-    @test typeof(dt[:, 2]) == NullableVector{Int}
-    @test typeof(dt[:, 3]) == NullableVector{Int}
+    @test typeof(dt[:, 1]) == Vector{?Int}
+    @test typeof(dt[:, 2]) == Vector{?Int}
+    @test typeof(dt[:, 3]) == Vector{?Int}
     @test all(isnull, dt[:, 1])
     @test all(isnull, dt[:, 2])
     @test all(isnull, dt[:, 3])
 
-    dt = DataTable([Nullable{Int}, Nullable{Float64}, Nullable{String}], 100)
+    dt = DataTable([?Int, ?Float64, ?String], 100)
     @test size(dt, 1) == 100
     @test size(dt, 2) == 3
-    @test typeof(dt[:, 1]) == NullableVector{Int}
-    @test typeof(dt[:, 2]) == NullableVector{Float64}
-    @test typeof(dt[:, 3]) == NullableVector{String}
+    @test typeof(dt[:, 1]) == Vector{?Int}
+    @test typeof(dt[:, 2]) == Vector{?Float64}
+    @test typeof(dt[:, 3]) == Vector{?String}
     @test all(isnull, dt[:, 1])
     @test all(isnull, dt[:, 2])
     @test all(isnull, dt[:, 3])
 
-    dt = DataTable([Nullable{Int}, Nullable{Float64}, Nullable{String}],
+    dt = DataTable([?Int, ?Float64, ?String],
                    [:A, :B, :C], 100)
     @test size(dt, 1) == 100
     @test size(dt, 2) == 3
-    @test typeof(dt[:, 1]) == NullableVector{Int}
-    @test typeof(dt[:, 2]) == NullableVector{Float64}
-    @test typeof(dt[:, 3]) == NullableVector{String}
+    @test typeof(dt[:, 1]) == Vector{?Int}
+    @test typeof(dt[:, 2]) == Vector{?Float64}
+    @test typeof(dt[:, 3]) == Vector{?String}
     @test all(isnull, dt[:, 1])
     @test all(isnull, dt[:, 2])
     @test all(isnull, dt[:, 3])
 
 
-    dt = DataTable([Nullable{Int}, Nullable{Float64}, Nullable{String}],
-                   [:A, :B, :C], [false, false, true],100)
-    @test size(dt, 1) == 100
-    @test size(dt, 2) == 3
-    @test typeof(dt[:, 1]) == NullableVector{Int}
-    @test typeof(dt[:, 2]) == NullableVector{Float64}
-    @test typeof(dt[:, 3]) == NullableCategoricalVector{Compat.UTF8String,UInt32}
-    @test all(isnull, dt[:, 1])
-    @test all(isnull, dt[:, 2])
-    @test all(isnull, dt[:, 3])
+    #FIXME: something in CategoricalArrays
```

Comment:
Ah, that's a missing constructor then. I think we will end up with a slightly different type, which would have element type `Union{CategoricalValue{T}, Null}` rather than `CategoricalValue{Union{T, Null}}`. That's a somewhat subtle difference, but I think it's better. If you're interested in discussing this, let's do it at https://github.com/JuliaData/CategoricalArrays.jl/pull/70.